### PR TITLE
Move async article from WebFu.

### LINKED ISSUE #4437: content: Move "Async functions" article from WebFu
* Use this template to request updates to existing content.
* Delete any fields that aren't relevant to your issue. 
* Update the title of this issue with a brief summary of the work to be done.

### What page(s) need to be updated?

https://developers.google.com/web/fundamentals/primers/async-functions

### Why is this update needed?

[Top-level wait](https://www.chromestatus.com/features/5767881411264512) is shipping in Chrome 89. The most logical place to put this is in the existing async functions article. (This feature has already been merged to the Chrome 89 branch, so there's no question about whether it will ship.)

(While we're at it, we should make reciprocal links between this and [JavaScript Promises](https://web.dev/promises/).)

### What's the deadline?

January 28

#### Is it a hard deadline?

Yes. It would be nice to have something to link to in the beta release blog. 

If yes, please explain why.

### PR DESCRIPTION
**Pull request by jpmedley**
Fixes #4437

Changes proposed in this pull request:

- Converts "Async functions: making promises friendly" from WebFu
- Adds reciprocal links between this and "JavaScript Promises: An introduction"

### CODE REVIEW

**Comment by jpmedley** [comment]
@jakearchibald, I've added a line to the first paragraph of the async article to go to the earlier article for info on Promises. There's no link from the basic promises article yet because I'm not sure where to put it. 

I have two questions about some of the content. What do you think about replacing the reference material at the bottom with a link to [what's on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)? Is the stuff about Generators still needed?

File: src/site/content/en/blog/async-functions/index.md
Comment (jakearchibald) [inline src/site/content/en/blog/async-functions/index.md:42]: Does this link work? It's `./promises` above.

File: src/site/content/en/blog/async-functions/index.md
Comment (jakearchibald) [inline src/site/content/en/blog/async-functions/index.md:183]:
```suggestion
I've shown you `async function() {}` already, but the `async` keyword can be
```

**Comment by jakearchibald** [comment]
> What do you think about replacing the reference material at the bottom with a link to [what's on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)?

If you mean https://web.dev/promises/#promise-api-reference, then I agree, this can be replaced with a link to MDN.

> Is the stuff about Generators still needed?

I think that can be removed, especially as it's covered much more succinctly in the async functions article.

> There's no link from the basic promises article yet because I'm not sure where to put it.

I think there are a few examples in the promises article that would be a lot simpler with async functions. Give me a shout when this PR lands and I'll go through the promise article and create links & notes to point to the async functions one.

**Comment by jpmedley** [comment]
>Give me a shout when this PR lands and I'll go through the promise article and create links & notes to point to the async functions one.

Do you mean after this is merged?

**Comment by jakearchibald** [comment]
Yes

**Comment by kaycebasques** [comment]
@jpmedley LMK when you and Jake are done collaborating and I'll do a TW review

**Comment by jpmedley** [comment]
@jakearchibald Sorry for the delay. I got sidetracked by a fire (not literal), then I got sick. PTAL.

File: src/site/content/en/blog/async-functions/index.md
Comment (robdodson) [inline src/site/content/en/blog/async-functions/index.md:336]: Netlify is failing to build because somewhere in this file, one of these `{: }` directives isn't working. Taking a quick glance I was wondering if it's this one because it doesn't have a space between the colon and the dot?

**Review by robdodson — CHANGES_REQUESTED** [review]

File: src/site/content/en/blog/async-functions/index.md
Comment (jpmedley) [reply src/site/content/en/blog/async-functions/index.md]: I tried adding a space. See if that works.

Comment (jpmedley) [reply src/site/content/en/blog/async-functions/index.md]: Everything seems to pass now. PTAL.

**Comment by jpmedley** [comment]
@jakearchibald Should we throw something in the await article about [top-level await](https://www.chromestatus.com/feature/5767881411264512) or do you want to do something else?

**Comment by jpmedley** [comment]
Ping on this. Beta release for this is getting close.

**Comment by jpmedley** [comment]
@jakearchibald Will you have a chance to look at this soon. I'd really appreciate it.

**Comment by kaycebasques** [comment]
I'll move forward with reviewing this because there's probably no major reason to block it any longer

File: src/site/content/en/blog/async-functions/index.md
Comment (kaycebasques) [inline src/site/content/en/blog/async-functions/index.md:81]: @jpmedley please rephrase to avoid "native"

File: src/site/content/en/blog/async-functions/index.md
Comment (kaycebasques) [inline src/site/content/en/blog/async-functions/index.md:172]:
```suggestion
[async iterators](https://github.com/tc39/proposal-async-iteration),
```
Please remove all these `{: .external}` things. They don't do anything on web.dev and they are breaking the build.

File: src/site/content/en/blog/async-functions/index.md
Comment (kaycebasques) [inline src/site/content/en/blog/async-functions/index.md:293]: Please replace https://web.dev/handbook/web-dev-components/#compare

File: src/site/content/en/blog/async-functions/index.md
Comment (kaycebasques) [inline src/site/content/en/blog/async-functions/index.md:308]: Please replace https://web.dev/handbook/web-dev-components/#compare

File: src/site/content/en/blog/async-functions/index.md
Comment (kaycebasques) [inline src/site/content/en/blog/async-functions/index.md:331]: Please move this so that it's the first section of the page

**Review by kaycebasques — CHANGES_REQUESTED** [review]
@jpmedley PTAL

**Comment by kaycebasques** [comment]
Status: waiting on @jpmedley to address my last round of feedback

File: src/site/content/en/blog/async-functions/index.md
Comment (jpmedley) [reply src/site/content/en/blog/async-functions/index.md:81]: @jakearchibald I don't know what non-native promises are.

File: src/site/content/en/blog/async-functions/index.md
Comment (jpmedley) [reply src/site/content/en/blog/async-functions/index.md]: Removed.

File: src/site/content/en/blog/async-functions/index.md
Comment (jpmedley) [reply src/site/content/en/blog/async-functions/index.md:293]: Done.

File: src/site/content/en/blog/async-functions/index.md
Comment (jpmedley) [reply src/site/content/en/blog/async-functions/index.md:308]: Done.

File: src/site/content/en/blog/async-functions/index.md
Comment (jpmedley) [reply src/site/content/en/blog/async-functions/index.md:331]: Changed the first line to:

>Async functions are enabled by default in Chrome, Edge, Firefox, and Safari, and they're quite frankly marvelous. 

Changed this paragraph to:

>Async functions are enabled by default in most major browsers. If your site still needs to support older browsers without async, here are a few workarounds.

**Comment by jpmedley** [comment]
PTAL

**Comment by robdodson** [comment]
It looks like this isn't stale. @mihajlija do you have the bandwidth to do a final review since Kayce is out?

**Comment by jpmedley** [comment]
@rachelandrew Are you still short on work?

**Comment by rachelandrew** [comment]
> @rachelandrew Are you still short on work?

I have things, but can rearrange priorities if needed!

File: src/site/content/en/blog/async-functions/index.md
Comment (jakearchibald) [inline src/site/content/en/blog/async-functions/index.md:81]:
```suggestion
Anything you `await` is passed through `Promise.resolve()`, so you can
safely `await` non-platform promises, such as those created by promise polyfills.
```

File: src/site/content/en/blog/async-functions/index.md
Comment (jakearchibald) [reply src/site/content/en/blog/async-functions/index.md:81]: Added suggestion below

**Comment by jpmedley** [comment]
@kaycebasques Can you please clear this before you leave.

**Comment by robdodson** [comment]
@jpmedley if the article is finished I can merge it as an admin. Would you like me to do that?

**Comment by jpmedley** [comment]
This is finished.

**Comment by robdodson** [comment]
@jpmedley do you need to add a redirect to webfu to point at this new article?

**Pull request merged**
